nsot/models.py

```diff
@@ -1182,6 +1182,15 @@ class Interface(Resource):
         verbose_name='Device', help_text='Unique ID of the connected Device.'
     )
 
+    # Cached hostname of the associated device
+    device_hostname = models.CharField(
+        max_length=255, null=False, blank=True, db_index=True,
```

Comment:
Please add `editable=False` to this field so that it doesn't show up in any forms automatically generated from the model object. This will also cause the field to become read-only in API serializers.

You'll need to also update the database migration file to include this flag.